net_handler/net_handler.c

```diff
@@ -117,6 +118,9 @@ int main() {
     }
     shm_init();
 
+    // TODO: Net Handler is in charge of regulating game states.
+    // Net Handler may not have this responsibility in the future.
+    start_gamestate_handler_thread();
```

Comment:
This looks a little ugly. Why have the `gamestate_filter.c` be in the `executor` folder if it is being imported into the `net_handler` only? Either start the thread from the `executor` or move the source file to the `net_handler` directory.

Reply:
It's also clunky because `studentapi.pyx` uses `filter_device_write_uid()` which is in `gamestate_filter.c` We're thinking of changing the implementation and bake the game state handler into `shm_wrapper.c`, but probably after this PR, since we want it done before this upcoming week.

I tried putting it in `executor`, but then it makes it awkward to test if the test file doesn't use a student code. (if there's no student code in a test case, then `executor` doesn't spawn)

Reply:
I have an additional idea for this after much thinking (check Slack) that we can discuss?